src/renderer/Renderer.ts

```diff
@@ -0,0 +1,171 @@
+import { drawAxes, DrawAxesProps } from './commands/drawAxes';
+import { drawObject, DrawObjectProps } from './commands/drawObject';
+
+import { mat4, vec4 } from 'gl-matrix';
+
+// tslint:disable-next-line:import-name
+import REGL = require('regl');
+
+// tslint:disable:no-unsafe-any
+
+/*
+ * A collection of the properties needed to render something using the default shader
+ */
+export interface RenderObject {
+    transform: REGL.Mat4;
+    vertices: REGL.Vec3[];
+    normals: REGL.Vec3[];
+    colors: REGL.Vec3[];
+    indices: number[];
+}
+
+/**
+ * Manages all scene information and is responsible for rendering it to the screen
+ */
+export class Renderer {
+    public readonly width: number;
+    public readonly height: number;
+    public readonly stage: HTMLDivElement;
+
+    private clearAll: () => void;
+    private clearDepth: () => void;
+    private drawObject: REGL.DrawCommand<REGL.DefaultContext, DrawObjectProps>;
+    private drawAxes: REGL.DrawCommand<REGL.DefaultContext, DrawAxesProps>;
+
+    private cameraTransform: mat4 = mat4.create();
+    private projectionMatrix: mat4 = mat4.create();
+    private ctx2D: CanvasRenderingContext2D;
+
+    constructor(width: number, height: number) {
+        this.width = width;
+        this.height = height;
+
+        // Create
```

Comment:
Create what?

Reply:
lol woops guess I never finished this comment